src/compiler/checker.rs

```diff
@@ -179,12 +240,31 @@ impl TypeChecker {
 
     fn create_intrinsic_type(&self, kind: TypeFlags, intrinsic_name: &str) -> BaseIntrinsicType {
         let type_ = self.create_type(kind);
-        let type_ = BaseIntrinsicType::new(type_);
+        let type_ = BaseIntrinsicType::new(type_, intrinsic_name.to_string());
         type_
     }
 
     fn type_to_string(&self, type_: &Type) -> String {
-        unimplemented!()
+        let writer = Rc::new(RefCell::new(create_text_writer("")));
```

Comment:
This was sort of messy/confusing, I need to pass this writer to `printer.write_node()` below "by reference" and then read its value (`.get_text()`) after that. So the printer can't take ownership of the writer. I tried passing a reference to the writer to the printer but wasn't getting that to work partly because the writer argument to `.write_node()` (which gets stored in the "state" of the printer) is of a trait type (`EmitTextWriter`?) so seemingly has to get passed/stored as a `dyn` trait object (I forget exactly what I ran into that wasn't working)

So ended up making the type it's stored as on the printer an `Rc<RefCell<...>>` but then here the caller has to be aware of that since the `Rc`-ness is the way that it's able to pass it "by reference" and still use it after